### PR TITLE
Allow evaluation of "top of set" and "bottom of set" ECL expressions

### PR DESCRIPTION
**Pull request by apeteri**

### CODE REVIEW

**Review by cmark — APPROVED** [review]
🥓 

Just a small note: the ECLRewriter logic can also be used to convert some constructs to another ones.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ecl/SnomedEclEvaluationRequest.java [hunk 1]
```diff
@@ -646,55 +647,40 @@ protected Class<?> getDocumentType() {
...
+			return toStringSet(snomedConcept.getAncestorIds());
 		} else {
-			super.addAncestorIds(concept, collection);
+			return toStringSet(snomedConcept.getStatedAncestorIds());
+		}
+	}
+	
+	private Set<String> toStringSet(final long[] idArray) {
+		if (idArray == null || idArray.length < 1) {
+			return newHashSet();
```
Comment (nagyo) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ecl/SnomedEclEvaluationRequest.java:677]: I'm just wondering if this should be mutable. Or would it be possible to use an immutable empty set here?

**Review by nagyo — APPROVED** [review]
LGTM!

**Review by cmark — APPROVED** [review]
🥓

**Pull request merged**